public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -74,7 +74,7 @@ function useDatasource(datasourceName: string) {
   useEffect(() => {
     getDatasourceSrv()
       .loadDatasource(datasourceName)
-      .then((datasource: CloudWatchDatasource) => setDatasource(datasource));
+      .then((datasource) => setDatasource(datasource as CloudWatchDatasource));
```

Comment:
this `as` seems pretty problematic and a a big opportunity to introduce bugs. Let me come back at this with maybe another approach.

Reply:
fair enough 👍

Reply:
Nothing really came to mind about a better approach for this, and I know too little about this part of grafana.

Sidenote - calling `loadDatasource` in "userland" plugin code (like this ConfigEditor) seems like a smell/wrong pattern to me? Only one other core plugin (mysql?) does it (and it types it to `any`  🤢 ), but I'm not sure about better alternatives. @sunker

Reply:
@joshhunt i've removed it from this PR for now anyway, so we can think about it and fix at a later date 👍